### PR TITLE
feat(ux): add "Most Recent" button = home

### LINKED ISSUE #66: Add a home button
Once you're on a tab, it's not obvious that you can click the logo to go back to the home page.
Maybe we could add a home button, or some other name? @wadhwamatic wdyt?

Screenshot 2023-04-21 at 4 25 18 PM

### PR DESCRIPTION
**Pull request by ericboucher**
Fixes #66 

@wadhwamatic @sifelix  maybe we could find a more telling word than `home`? Something more descriptive of what the page shows. How about `Latest Reports` `Latest Submissions`?

Screenshot 2023-04-27 at 5 14 15 PM

### CODE REVIEW

File: apps/frontend/components/NavBar/NavMenuContent.tsx [hunk 1]
```diff
@@ -67,6 +68,24 @@ export const NavMenuContent = (): JSX.Element => {
         </IconButton>
       </Link>
       <List>
+        <ListItem key="home">
```
Comment (ericboucher) [inline apps/frontend/components/NavBar/NavMenuContent.tsx:71]: @sifelix there is a lot of duplicative code in this file

**Review by sifelix — APPROVED** [review]

**Comment by sifelix** [comment]
> Fixes #66
> 
> @wadhwamatic @sifelix maybe we could find a more telling word than `home`? Something more descriptive of what the page shows. How about `Latest Reports` `Latest Submissions`?
> 
> Screenshot 2023-04-27 at 5 14 15 PM

I don't know what would be the best. It is also the Home page

**Comment by ericboucher** [comment]
@wadhwamatic how about `Recent`, with this icon? https://mui.com/material-ui/material-icons/?query=clock&selected=History
Screenshot 2023-04-28 at 6 49 24 PM

**Comment by ericboucher** [comment]
@sifelix technically these are "forms" not reports, correct? Can we just say `entry date` instead of `report entry date` which I think is misleading?

**Pull request merged**